golang/testsuite/networks_impl/test_network.go

```diff
@@ -213,3 +234,61 @@ func createDatastoreConfigFileInServiceDirectory(network *TestNetwork, sharedDir
 
 	return configFileFilePath, nil
 }
+
+func newDatastoreClient(datastoreIp string) (datastore_rpc_api_bindings.DatastoreServiceClient, func() error, error) {
+	datastoreURL := fmt.Sprintf(
+		"%v:%v",
+		datastoreIp,
+		datastorePort,
+	)
+
+	conn, err := grpc.Dial(datastoreURL, grpc.WithInsecure())
+	if err != nil {
+		return nil, nil, stacktrace.Propagate(err, "An error occurred dialling the datastore container via its URL")
+	}
+
+	datastoreServiceClient := datastore_rpc_api_bindings.NewDatastoreServiceClient(conn)
+
+	return datastoreServiceClient, conn.Close, nil
+}
+
+func newExampleAPIServerClient(exampleAPIServerIp string) (example_api_server_rpc_api_bindings.ExampleAPIServerServiceClient, func() error, error) {
+	exampleAPIServerURL := fmt.Sprintf(
+		"%v:%v",
+		exampleAPIServerIp,
+		apiServicePort,
+	)
+
+	conn, err := grpc.Dial(exampleAPIServerURL, grpc.WithInsecure())
+	if err != nil {
+		return nil, nil, stacktrace.Propagate(err, "An error occurred dialling the example API server container via its URL")
+	}
+
+	exampleAPIServerClient := example_api_server_rpc_api_bindings.NewExampleAPIServerServiceClient(conn)
+
+	return exampleAPIServerClient, conn.Close, nil
+}
+
+func waitForHealthy(ctx context.Context, client GRPCAvailabilityChecker, retries uint32, retriesDelayMilliseconds uint32) error {
+
+	var (
+		emptyArgs = &empty.Empty{}
+		err       error
+	)
+
+	for i := uint32(0); i < retries; i++ {
+		_, err = client.IsAvailable(ctx, emptyArgs)
+		if err == nil {
```

Comment:
Same comment as in https://github.com/kurtosis-tech/kurtosis-cli/pull/61 , about not needing the `if err != nil` check down below by changing this to a `return nil`